### PR TITLE
Team7

### PR DESCRIPTION
**Pull request by moinbukhari**
# Summary

Ran make run, lint and format.

First PR, that has multiple types of agents in progress. Agent 1 is in working condition and feeds off the 5 common personality traits. At the moment it is basic but will get more intelligent.

### CODE REVIEW

File: pkg/agents/team7/agent1/agent.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+	neuroticism int
+}
+
+type CustomAgent7 struct {
+	*infra.Base
+	personality team7Personalities
+	greediness  int
+	kindness    int
+	daysAlive   int
```
Comment (jzzheng22) [inline pkg/agents/team7/agent1/agent.go:60]:
```suggestion
```

Comment (jzzheng22) [reply pkg/agents/team7/agent1/agent.go:60]: This was added as a field to base agent as of last night.

File: pkg/agents/team7/agent1/agent.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+		rand.Seed(time.Now().UTC().UnixNano())
+		a.greediness = 100 - a.personality.agreeableness
+		a.kindness = a.personality.agreeableness
+	}
+
+	a.Log("Team7Agent1 reporting status:", infra.Fields{"floor": a.Floor(), "hp": a.HP(), "greed": a.greediness, "kind": a.kindness, "aggr": a.personality.agreeableness})
+
+	//Check if new day has started
+	if a.HP() < a.prevHP {
+		a.daysAlive++
```
Comment (jzzheng22) [inline pkg/agents/team7/agent1/agent.go:98]:
```suggestion
```

File: pkg/agents/team7/agent1/agent.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+	if a.daysAlive == 0 {
+		rand.Seed(time.Now().UTC().UnixNano())
+		a.greediness = 100 - a.personality.agreeableness
+		a.kindness = a.personality.agreeableness
+	}
+
+	a.Log("Team7Agent1 reporting status:", infra.Fields{"floor": a.Floor(), "hp": a.HP(), "greed": a.greediness, "kind": a.kindness, "aggr": a.personality.agreeableness})
+
+	//Check if new day has started
+	if a.HP() < a.prevHP {
```
Comment (jzzheng22) [inline pkg/agents/team7/agent1/agent.go:97]: This is not a guarantee that a new day has started. It is possible for `a.HP() >= a.prevHP` to be true.

File: pkg/agents/team7/agent1/agent.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+			a.kindness = 100
+		}
+
+		if a.greediness < 0 {
+			a.greediness = 0
+		} else if a.greediness > 100 {
+			a.greediness = 100
+		}
+
+		var foodtotake food.FoodType = food.FoodType(100 - a.kindness + a.greediness)
```
Comment (jzzheng22) [inline pkg/agents/team7/agent1/agent.go:151]:
```suggestion
		foodtotake = food.FoodType(100 - a.kindness + a.greediness)
```

File: pkg/agents/team7/agent1/agent.go [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+// 	currentHP := a.HP()
+// 	//currentFloor := a.Floor()
+// 	//currentAvailFood := a.CurrPlatFood()
+
+// 	var foodtotake food.FoodType = food.FoodType(100 - currentHP)
+// 	if foodtotake != 0 {
+// 		a.TakeFood(foodtotake)
+// 	}
+
+// }
```
Comment (jzzheng22) [inline pkg/agents/team7/agent1/agent.go:188]:
```suggestion
```

File: pkg/agents/team7/agent2/agent.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/food"
+)
+
+type CustomAgent1 struct {
+	*infra.Base
+	// new params
+}
```
Comment (jzzheng22) [inline pkg/agents/team7/agent2/agent.go:11]:
```suggestion
type CustomAgent2 struct {
	*infra.Base
	// new params
}
```
You should change all instances of this.

File: pkg/agents/team7/agent2/agent.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+	Y := 5
+	X := 15
+
+	//currentAvailFood := a.CurrPlatFood()
+
+	if F < Y {
+		foodtotake := food.FoodType(G - currentHP)
+		if foodtotake != 0 {
+			a.TakeFood(foodtotake)
+		}
```
Comment (jzzheng22) [inline pkg/agents/team7/agent2/agent.go:41]: `a.TakeFood` already checks for 0. You don't need to handle it here.

File: pkg/agents/team7/agent2/agent.go [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+	currentHP := a.HP()
+
+	H := 60 //Healthy HP
+	G := 40 //Green HP
+	// W := 10 //Weak HP
+	C := 5 //Critial HP
+
+	F := a.Floor() //currentFloor := a.Floor()
+	Y := 5
+	X := 15
```
Comment (jzzheng22) [inline pkg/agents/team7/agent2/agent.go:33]: Single letter variable names should be avoided.

File: pkg/agents/team7/agent3/agent.go [hunk 9]
```diff
@@ -0,0 +1,33 @@
+package team7agent3
+
+import (
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+)
+
+type CustomAgent1 struct {
```
Comment (jzzheng22) [inline pkg/agents/team7/agent3/agent.go:7]:
```suggestion
type CustomAgent3 struct {
```
This should be changed for all instances.

File: pkg/agents/team7/agent3/agent.go [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+func (a *CustomAgent1) Run() {
+	a.Log("Team 7 Agent 3 reporting status:", infra.Fields{"floor": a.Floor(), "hp": a.HP()})
+
+	// UserID := a.ID()
+	// currentHP := a.HP()
+	// currentAvailFood := a.CurrPlatFood()
+
+	// currentFloor := a.Floor()
+	// a.prevFloors = append(a.prevFloors, currentFloor)
+
```
Comment (jzzheng22) [inline pkg/agents/team7/agent3/agent.go:32]: Why do you have this agent???

File: pkg/agents/team7/agent4/agent.go [hunk 11]
```diff
@@ -0,0 +1,62 @@
+package team7agent4
+
+import (
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+)
+
+type CustomAgent1 struct {
```
Comment (jzzheng22) [inline pkg/agents/team7/agent4/agent.go:7]:
```suggestion
type CustomAgent4 struct {
```
This should be changed for all instances.

File: pkg/agents/team7/agent4/agent.go [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+	// 	a.prevFloors = append(a.prevFloors, currentFloor)
+	// }
+	if a.orderPrevFloors[len(a.orderPrevFloors)-1] != currentFloor {
+		a.orderPrevFloors = append(a.orderPrevFloors, currentFloor)
+	}
+
+	if a.currentDayonFloor == 0 {
+
+	}
+
```
Comment (jzzheng22) [inline pkg/agents/team7/agent4/agent.go:51]:
```suggestion
```

File: pkg/agents/team7/agent4/agent.go [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+	if currentHP < a.lastHP {
+		a.lastHP = currentHP
+		a.currentDayonFloor++
+	}
+
+	currentFloor := a.Floor()
+	// if(a.prevFloors[len(a.prevFloors) - 1] != currentFloor){
+	// 	a.prevFloors = append(a.prevFloors, currentFloor)
+	// }
```
Comment (jzzheng22) [inline pkg/agents/team7/agent4/agent.go:43]:
```suggestion
```

File: pkg/agents/team7/agent4/agent.go [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+func (a *CustomAgent1) Run() {
+	a.Log("Team 7 Agent 4 reporting status:", infra.Fields{"floor": a.Floor(), "hp": a.HP()})
+
+	// UserID := a.ID()
+	currentHP := a.HP()
+	currentAvailFood := a.CurrPlatFood()
+
+	if currentHP < a.lastHP {
```
Comment (jzzheng22) [inline pkg/agents/team7/agent4/agent.go:35]: Not a guarantee that day has passed.

File: pkg/agents/team7/agent4/agent.go [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+		Base:              baseAgent,
+		currentDayonFloor: 1,
+		lastHP:            100,
+	}, nil
+}
+
+func (a *CustomAgent1) Run() {
+	a.Log("Team 7 Agent 4 reporting status:", infra.Fields{"floor": a.Floor(), "hp": a.HP()})
+
+	// UserID := a.ID()
```
Comment (jzzheng22) [inline pkg/agents/team7/agent4/agent.go:31]:
```suggestion
```

File: pkg/agents/team7/agent5/agent.go [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+	//create other parameters
+	return &CustomAgent1{
+		Base: baseAgent,
+	}, nil
+}
+
+func (a *CustomAgent1) Run() {
+	a.Log("Team 7 Agent 5 reporting status:", infra.Fields{"floor": a.Floor(), "hp": a.HP()})
+
+	// UserID := a.ID()
```
Comment (jzzheng22) [inline pkg/agents/team7/agent5/agent.go:26]: Why do you have this agent??

File: pkg/messages/ackMessage.go [hunk 17]
```diff
@@ -1,18 +1,18 @@
...
+// 	}
+// 	return msg
+// }
 
-func (msg AckMessage) MessageType() string {
-	return "AckMessage"
-}
+// func (msg AckMessage) MessageType() string {
+// 	return "AckMessage"
+// }
```
Comment (jzzheng22) [inline pkg/messages/ackMessage.go:18]: This file has been deleted. Do not add it back.

**Review by jzzheng22 — CHANGES_REQUESTED** [review]
Why has the `.ionide/symbolCache.db` file been included??

Please do not use merge commits before creating a PR. Learn to rebase - this PR was difficult to review since files which were not relevant and were unchanged were included.

Fix your merge conflicts.

File: pkg/agents/team7/agent5/agent.go (hunk 16 above)
Comment (moinbukhari) [reply pkg/agents/team7/agent5/agent.go:26]: someone else was working on it but I have deleted it for now

**Comment by JaafarRammal** [comment]
@moinbukhari you can run `make lint` to address pipeline issues. See https://github.com/SOMAS2021/SOMAS2021/runs/4659135511?check_suite_focus=true#step:4:75 for details

File: config.json [hunk 18]
```diff
@@ -1,17 +1,24 @@
...
+    "Team7Agent5": 0,
+
     "AgentHP": 100,
     "AgentsPerFloor": 1,
-    "TicksPerFloor": 10,
+    "TicksPerFloor": 1,
     "SimDays": 8,
-    "ReshuffleDays": 1,
+    "ReshuffleDays": 3,
+
```
[the remark this answers is not in the document]
Comment (moinbukhari) [reply config.json]: done

File: pkg/agents/team7/agent1/agent.go [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+		prevHP:     100,
+		prevFloors: []int{},
+	}, nil
+}
+
+func (a *CustomAgent7) Run() {
+
+	//initialise greediness and kindness
+	if a.daysAlive == 0 {
+		rand.Seed(time.Now().UTC().UnixNano())
```
[the remark this answers is not in the document]
Comment (moinbukhari) [reply pkg/agents/team7/agent1/agent.go]: done

File: pkg/agents/team7/agent1/agent.go (hunk 1 above)
Comment (moinbukhari) [reply pkg/agents/team7/agent1/agent.go]: using AGe

File: pkg/agents/team7/agent1/agent.go (hunk 2 above)
Comment (moinbukhari) [reply pkg/agents/team7/agent1/agent.go]: done

File: pkg/agents/team7/agent1/agent.go (hunk 3 above)
Comment (moinbukhari) [reply pkg/agents/team7/agent1/agent.go]: fixed it with new code

File: config.json [hunk 20]
```diff
@@ -1,17 +1,19 @@
...
-    "Team6Agents": 2,
-    "RandomAgents": 2,
+    "Team1Agents": 0,
+    "Team2Agents": 0,
+    "Team3Agents": 0,
+    "Team4Agents": 0,
+    "Team5Agents": 0,
+    "Team6Agents": 0,
+    "RandomAgents": 0,
+    "Team7Agent1": 10,
```
Comment (jzzheng22) [inline config.json:10]: You still haven't changed this back to the original values.

Also please move `RandomAgent` to the last item in the list please.

File: pkg/agents/team7/agent1/agent.go [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+	}, nil
+}
+
+func (a *CustomAgent7) Run() {
+
+	//initialise greediness and kindness
+	if a.Age() == 0 {
+		a.greediness = 100 - a.personality.agreeableness
+		a.kindness = a.personality.agreeableness
+	}
```
Comment (jzzheng22) [inline pkg/agents/team7/agent1/agent.go:92]: Why would you not put this in the constructor?

File: pkg/agents/team7/agent1/agent.go [hunk 22]
```diff
@@ -0,0 +1,167 @@
...
+			a.greediness = 100
+		}
+
+		foodtotake := food.FoodType(100 - a.kindness + a.greediness)
+
+		//When platform reaches our floor and we haven't tried to eat, then try to eat
+		if a.CurrPlatFood() != -1 && !a.seenPlatform {
+			a.foodEaten = a.TakeFood(foodtotake)
+			if a.foodEaten > 0 {
+				a.Log("Team 7 has seen the platform:", infra.Fields{"foodEaten": a.foodEaten, "health": a.HP()})
```
Comment (jzzheng22) [inline pkg/agents/team7/agent1/agent.go:150]: Your `foodEaten` could be 0 even when you have seen the platform

File: pkg/config/configparameters.go [hunk 23]
```diff
@@ -20,6 +20,7 @@ type ConfigParameters struct {
 	Team5Agents    int           `json:"Team5Agents"`
 	Team6Agents    int           `json:"Team6Agents"`
 	RandomAgents   int           `json:"RandomAgents"`
+	Team7Agent1    int           `json:"Team7Agent1"`
```
Comment (jzzheng22) [inline pkg/config/configparameters.go:23]:
```suggestion
	Team7Agent1    int           `json:"Team7Agent1"`
	RandomAgents   int           `json:"RandomAgents"`
```

File: pkg/config/configparameters.go [hunk 24]
```diff
@@ -60,7 +61,7 @@ func LoadParamFromJson(path string) (ConfigParameters, error) {
 		return tempParameters, err
 	}
 	//appending the sizes of the agents to the array
-	tempParameters.NumOfAgents = append(tempParameters.NumOfAgents, tempParameters.Team1Agents, tempParameters.Team2Agents, tempParameters.Team3Agents, tempParameters.Team4Agents, tempParameters.Team5Agents, tempParameters.Team6Agents, tempParameters.RandomAgents)
+	tempParameters.NumOfAgents = append(tempParameters.NumOfAgents, tempParameters.Team1Agents, tempParameters.Team2Agents, tempParameters.Team3Agents, tempParameters.Team4Agents, tempParameters.Team5Agents, tempParameters.Team6Agents, tempParameters.RandomAgents, tempParameters.Team7Agent1)
```
Comment (jzzheng22) [inline pkg/config/configparameters.go:64]:
```suggestion
	tempParameters.NumOfAgents = append(tempParameters.NumOfAgents, tempParameters.Team1Agents, tempParameters.Team2Agents, tempParameters.Team3Agents, tempParameters.Team4Agents, tempParameters.Team5Agents, tempParameters.Team6Agents, tempParameters.Team7Agent1, tempParameters.RandomAgents)
```

File: pkg/simulation/simAgent.go [hunk 25]
```diff
@@ -28,7 +29,7 @@ func (sE *SimEnv) generateInitialAgents(t *infra.Tower) {
 
 func (sE *SimEnv) createNewAgent(tower *infra.Tower, i, floor int) {
 	sE.Log("Creating new agent")
-	abs := []AgentNewFunc{agent1.New, agent2.New, team3.New, team4EvoAgent.New, team5.New, team6.New, randomAgent.New}
+	abs := []AgentNewFunc{agent1.New, agent2.New, team3.New, team4EvoAgent.New, team5.New, team6.New, randomAgent.New, team7agent1.New}
```
Comment (jzzheng22) [inline pkg/simulation/simAgent.go:32]:
```suggestion
	abs := []AgentNewFunc{agent1.New, agent2.New, team3.New, team4EvoAgent.New, team5.New, team6.New, team7agent1.New, randomAgent.New}
```

**Review by jzzheng22 — CHANGES_REQUESTED** [review]
You still haven't removed the `.ionide/symbolCache.db` file. Remove it please.

File: pkg/agents/team7/agent1/agent.go (hunk 21 above)
Comment (moinbukhari) [reply pkg/agents/team7/agent1/agent.go:92]: How do I use values that have just been initialised in a struct, 
basically: kindness = this.personality.agreeableness does not work

File: pkg/agents/team7/agent1/agent.go (hunk 22 above)
Comment (moinbukhari) [reply pkg/agents/team7/agent1/agent.go:150]: I have added an else statement here, not sure if that is the issue you are referring to

File: pkg/agents/team7/agent1/agent.go (hunk 21 above)
Comment (jzzheng22) [reply pkg/agents/team7/agent1/agent.go]: See Slack message

**Review by jzzheng22 — APPROVED** [review]

**Pull request merged**